plugins/ida/python/propagate_taint_bap_hexrays.py

```diff
@@ -0,0 +1,121 @@
+bap_color = {
+    'black'  : 0x000000,
+    'red'    : 0xCCCCFF,
+    'green'  : 0x99FF99,
+    'yellow' : 0xC2FFFF,
+    'blue'   : 0xFFB2B2,
+    'magenta': 0xFFB2FF,
+    'cyan'   : 0xFFFFB2,
+    'white'  : 0xFFFFFF,
+    'gray'   : 0xEAEAEA,
+}
+
+
+class BAP_Taint_Pseudocode(idaapi.plugin_t):
+
+    def autocolorize_function(self, cfunc):
+        simplelinevec = cfunc.get_pseudocode()
+
+        def ea_from_addr_tag(addr_tag):
+            return cfunc.treeitems.at(addr_tag).ea
+
+        for simpleline in simplelinevec:
+            self.color_line(simpleline, bap_color['gray'])
+                # Ready to be painted over by other colors
+            self.autocolorize_line(simpleline, ea_from_addr_tag)
+
+    def color_line(self, simpleline, color):
+        simpleline.bgcolor = color
+
+    def get_new_color(self, current_color, ea):
+        coloring_order = [
+            bap_color[c] for c in [
+                'gray',
+                'white',
+                'red',
+                'yellow',
+            ]
+        ]
+
+        BGR_MASK = 0xffffff
+
+        ea_color = idaapi.get_item_color(ea)
+
+        assert(ea_color & BGR_MASK in coloring_order)
```

Comment:
Hmm.. Good catch.. The `assert` would be triggered and we don't want it like that. Will make a modification and push that commit here.